### ISSUE TITLE
support aliyun ecs discovery

### ISSUE THREAD

**Issue by KeyOfSpectator** [prometheus/prometheus#11931]
### Proposal

Hi~ We are alibaba aliyun team, we have a aliyun ecs discovery implement.
https://github.com/AliyunContainerService/prometheus/tree/master/discovery/ecs
Can we take a merge-request of our commit to master branch?

- pressure test: this implement support pressure test with 1K+ ECS, and every ECS have 20 tags, support frequency ECS scale scenario.
- support filter with ECS label.
- use scrolling API to discovery ecs to support frequency ECS scale change scenario.

**Comment by roidelapluie** [comment]
A quick glance at the link shows me that we could accept it if there are enough tests and if we do not take configuration and secrets from environment variables. We'd also need an alibaba account to test this.

**Comment by KeyOfSpectator** [comment]
@roidelapluie 
ok~THX~
I will make a merge request with test case, and refine comments and README.
We can provide alibaba account for testing.

**Comment by KeyOfSpectator** [comment]
WIP, I m trying to make it out in this month.

> Is there any progress? I am very looking forward to the pr being merged

**Comment by beorn7** [comment]
Hello from the bug scrub.

@roidelapluie what's the current state of this?

We discussed the general state of SD recently (umbrella issue to be filed soon), as we see an increasing amount of issues maintaining all the SDs we currently have, not to speak about adding more. We need clear ownership for every mechanisms, so this here would need on owner for starters.

Alternatively, there is always the option of providing an external solution via HTTP SD.

**Comment by KeyOfSpectator** [comment]
> Hello from the bug scrub.
> 
> @roidelapluie what's the current state of this?
> 
> We discussed the general state of SD recently (umbrella issue to be filed soon), as we see an increasing amount of issues maintaining all the SDs we currently have, not to speak about adding more. We need clear ownership for every mechanisms, so this here would need on owner for starters.
> 
> Alternatively, there is always the option of providing an external solution via HTTP SD.

Apologies for the delayed response.

Regarding the current status of this issue, our team intends to submit an official Pull Request (PR) for the Alibaba Cloud ECS service discovery mechanism in the near future, specifically sometime within July.

Due to the stringent requirements for stability at the infrastructure layer, there is indeed a need for a service discovery approach beyond just relying on HTTP Service Discovery (HTTP SD) for Alibaba Cloud ECS instances.

Moving forward, it is proposed that our Alibaba Cloud team takes on the ongoing maintenance of this service discovery mechanism. Consequently, it is imperative that we ensure the official release version offers a mature SLA and is thoroughly covered by UT.

**Comment by beorn7** [comment]
This sounds very promising, thank you very much.

@simonpasquier FYI, this should be mentioned in the planned umbrella issue.

@roidelapluie I think you would be in the best position to coordinate with Alibaba folks how to integrate their code and onboard them as maintainers. It would be great if you could let us know if you could do that or not.

**Comment by roidelapluie** [comment]
Yes, I am waiting for more details and a proper PR from alibaba.

**Comment by KeyOfSpectator** [comment]
> Yes, I am waiting for more details and a proper PR from alibaba.
@roidelapluie 
Hi, sorry for the late. 
We have already make a PR: #15212 . PTAL.
And we made a pressure test report, published in the blog channel. [prometheus aliyun ecs sd pressure test report](https://www.alibabacloud.com/blog/the-pressure-test-of-prometheus-on-alibaba-cloud-ecs-service-discovery-feature_601694)
